// Copyright 2012 Google Inc. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// Protocol buffer for storing and serializing sentences, documents,
// collections of documents and summaries.

syntax = "proto2";

package topicsum;

// A sentence inside a document.
//
// Next Id: 10
message Sentence {
  //Identifier for the sentence
  optional string id = 8;

  // The raw content of the sentence, without any tokenization nor normalization
  // performed.
  optional string raw_content = 1;

  // A token in the sentence.
  repeated string token = 2;

  // Part-of-speech tags for the tokens. If this field is populated, its length
  // has to be the same as the length of the token field.
  repeated string pos_tag = 5;

  // Prior score of this sentence before summarization.
  optional float prior_score = 9;

  // Score of this sentence, if it was scored to appear in a summary.
  optional float score = 3;

  // When this sentence is included in an extractive summary, this field
  // contains the id of the document from which it was extracted.
  optional string source_document = 4;
}

// A full document.
//
// Generated summaries are also stored as documents.
//
// Next Id: 14
message Document {
  // Identifier for the document, e.g. URL if it is a web document.
  optional string id = 9;

  // Title of the document.
  optional string title = 1;

  // Description attribute of basic HTML meta element, commonly available in the
  // news articles, and identical to the RSS feed of major news sources,
  // providing a short summary of the news article.
  optional string description = 13;

  // Raw content of the document, without any sentence splitting or
  // normalization performed.
  optional string raw_content = 2;

  // Publication date of the document, in seconds since epoch.
  optional int64 publication_date = 3;

  // Sentences (usually obtained by sentence splitting the raw content).
  repeated Sentence sentence = 4;

  // Total number of sentence tokens.
  optional int64 num_tokens = 6;

  // Title sentences (usually obtained by sentence splitting the title).
  repeated Sentence title_sentence = 11;

  // Total number of title sentence tokens.
  optional int64 num_title_tokens = 12;

  // Debug information generated by the document builder.
  optional string debug_string = 7;

  // This flag needs to be set to true only if you want to produce an update
  // summary. An update summary is produced from a collection mixing 'update'
  // (dual) and 'base' documents. It only summarizes the new information in
  // 'update' documents assuming that the information from 'base' documents is
  // already known. If you just want to summarize a collection of document do
  // not touch this flag.
  optional bool dual = 8 [default=false];
}

// A document collection.
//
// Next Id: 5
message DocumentCollection {
  // Collection identifier.
  optional string id = 1;

  // List of documents in this collection.
  repeated Document document = 2;

  // Aggregate number of sentences in all the documents.
  optional int64 num_sentences = 3;
}
